### ISSUE TITLE
Button: Gradient backgrounds applied in Global Styles cannot be overridden with solid colors at the block level

### ISSUE THREAD

**Issue by ndiego** [WordPress/gutenberg#67055]
### Description

If you apply a gradient background to Button blocks in Global Styles → Blocks → Button (or theme.json), users cannot override this gradient background with a solid background color at the block level. They can override it with another gradient background, which makes sense from a technical standpoint but is very unintuitive from a user perspective. You should be able to set the background at the block level, whether you want to apply a gradient or solid color.

### Step-by-step reproduction instructions

1. Set a gradient background on the Button block in Global Styles and save.
2. Open a new post and add the Buttons block, and add a Button.
3. Confirm that the Button is displaying the gradient background set in Global Styles.
4. Try setting a block-level solid background color. Confirm nothing changes. 
5. Now add a block-level gradient background color. Confirm that the new gradient is applied. 

### Screenshots, screen recording, code snippet

| Global Styles | Block-level (Solid) | Block-level (Gradient) |
|-|-|-|
|![Image](https://github.com/user-attachments/assets/19d9d061-7da4-423b-bae9-4de96d00111e)|![Image](https://github.com/user-attachments/assets/1b69342c-1b2b-4d77-b96c-6c051e23164f)|![Image](https://github.com/user-attachments/assets/826113d9-1f99-486e-959d-dd7d3541663c)|

### Environment info

- WordPress 6.7
- With and without Gutenberg

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by im3dabasia** [comment]
I’ve tested this issue and can confirm it’s a bug. A gradient background set in Global Styles cannot be overridden with a solid color at the block level.

**Comment by carolinan** [comment]
Could we solve this everywhere instead of for one block?
https://github.com/WordPress/gutenberg/issues/47391
https://github.com/WordPress/gutenberg/issues/42105

**Comment by yogeshbhutkar** [comment]
> Could we solve this everywhere instead of for one block?
> [#47391](https://github.com/WordPress/gutenberg/issues/47391)

Thanks for the suggestion @carolinan, I'll try to solve it everywhere and update the PR.

**Comment by yogeshbhutkar** [comment]
@carolinan, The default gradient override bug has now been fixed for all the blocks in the latest patch of the PR.

**Comment by carolinan** [comment]
@yogeshbhutkar I don't think that adding a new class name to a block when there is a background color but no background gradient is the ideal solution.
- This class name is present on the block even if there was no gradient set in the global styles.
- This class name is present on the block even if the theme disables gradients.
- The CSS added to the class name breaks the background image setting (Tested with the group block)

**Comment by yogeshbhutkar** [comment]
Thank you for pointing that out, @carolinan. I believe I agree with your perspective on the class names approach. Could you suggest any alternative or improved methodologies that I could implement?

**Comment by carolinan** [comment]
I don't have a solution. There are past discussions here: https://github.com/WordPress/gutenberg/issues/32787

**Comment by Andrew-Starr** [comment]
When changing the background (either color or gradient) at the individual block, the `has-background` class is added to the block.

I think this is too broad a definition to use this class name for both color and gradient. Couldn't a separate class name of `has-gradient` be used for gradient background and keep `has-background` only for the solid background color?

If there was this separation then the global styles CSS could be changed to something like `:not(.has-background)` when applying a gradient to a block at the global styles level, and the global gradient could be overridden with a solid color in an individual block.

For example, the heading block is currently like this:
```
:root :where(.wp-block-heading) {
  background: var(--wp--preset--gradient--blush-light-purple);
}
```

If it was something like the following, it could be overridden with a solid color:
```
:root :where(.wp-block-heading:not(.has-background)) {
  background: var(--wp--preset--gradient--blush-light-purple);
}
```

**Comment by t-hamano** [comment]
Closing as duplicate of https://github.com/WordPress/gutenberg/issues/69634
